### PR TITLE
replace tiller grpc api error with grpc.Status

### LINKED ISSUE #2894: tiller api return coded error
use  [grpc status](https://github.com/grpc/grpc-go/blob/master/status/status.go) to return coded error for tiller grpc api

like we can change error define in [release_server.go](https://github.com/kubernetes/helm/blob/master/pkg/tiller/release_server.go)
```golang
var (
	// errMissingChart indicates that a chart was not provided.
	errMissingChart = errors.New("no chart provided")
        ...
)
```

into

```golang
import (
       "google.golang.org/grpc/status"
       "google.golang.org/grpc/codes"
)
    
var (
       errMissingChart = status.Errorf(codes.NotFound, "chart not found")
       ...
)
```

and the tiller client side can process the coded error like
```golang

resp, err := client.GetReleaseContent(ctx, req)
if err != nil {
    if s, ok := status.FromError(); ok {
        // check s.Code()  
    }
}

```
this won't change the current tiller grpc api, but will return a coded error for client program to process.

I can open a pr to change the related tiller error into grpc status.Status error.

@technosophos  what do you think?

### PR DESCRIPTION
**Pull request by weiwei04**
fix #2894

### CODE REVIEW

**Comment by bacongobbler** [comment]
I'm not as familiar with the grpc side of things in tiller, but from what I recall Kubernetes is on an older version of grpc (1.2.1) hence why we need to stick with that.

Anyone else more familiar on why we're still using 1.2.1 or are we good to move forward with 1.3.0? If that's the case maybe we should bump this all the way up to v1.6.0.

File: pkg/tiller/release_server.go [hunk 1]
```diff
@@ -29,7 +28,10 @@ import (
 	"k8s.io/client-go/discovery"
 	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
 
+	//"google.golang.org/grpc/codes"
```
Comment (thomastaylor312) [inline pkg/tiller/release_server.go]: Looks like these aren't needed anymore

**Comment by thomastaylor312** [comment]
I think most of our gRPC is just from the tiller side, I don't know if there is a transitive dependency problem though. @weiwei04 what k8s versions did you test this against?

**Comment by weiwei04** [comment]
I've deployed this modified tiller on our 1.7.3 kubernetes cluster :)

**Comment by thomastaylor312** [comment]
@weiwei04 Have you tried with the most up-to-date version of grpc? I agree with @bacongobbler that if we are going through the trouble of upgrading it, we may as well upgrade it all the way

**Comment by bacongobbler** [comment]
As per [the announcement on the mailing list](https://lists.cncf.io/g/cncf-helm/topic/last_v2_feature_release/34301238), Helm 2.15.0 is the last release where we will be merging new features for Helm 2. Because of that, I'm closing this PR due to inactivity. Since the gRPC API has been removed as part of Tiller's removal in Helm 3, it doesn't appear to make sense to port this over to Helm 3. Thanks!

**Pull request closed without merging**